qemu_build

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+BASENAME=bitnami/minideb
+pub_key_dir="$(mktemp -d)"
+
+do_sudo() {
+  if [[ "0" == "$(id --user)" ]]; then
+    "$@"
+  else
+    sudo "$@"
+  fi
+}
+
+do_ssh() {
+  ssh -o "StrictHostKeyChecking=no" -o "UserKnownHostsFile=/dev/null" root@localhost -t -p 5555 -i "$pub_key_dir/id_rsa" "$@"
+}
+
+finish() {
+  echo "Shutting down QEMU..."
+
+  n=0
+  until [ "$n" -ge 15 ]
+  do
+     do_ssh "true" && break
+     n=$((n+1))
+     sleep 30
+  done
+
+  do_ssh "poweroff" || true
+
+  sleep 5
+  n=0
+  until [ "$n" -ge 5 ]
+  do
+     kill -0 "$PID" && break
+     n=$((n+1))
+     sleep 5
+  done
+
+  kill -9 "$PID" || true
+  rm -f "$IMAGE_FILE" "$PIDFILE"
+}
+
+if [[ ! -f /etc/debian_version ]]; then
+  echo "minideb can currently only be built on debian based distros, aborting..."
+  exit 1
+fi
+
+if [ -z "$1" ]; then
+    echo "You must specify the dist to build"
+    exit 1
+fi
+
+DIST=$1
+PLATFORM=${2:-amd64}
+
+while do_sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
+    sleep 1
+done
+
+do_sudo apt-get update
+do_sudo apt-get install -y qemu-kvm libvirt-bin qemu-utils genisoimage virtinst curl rsync qemu-system-x86 qemu-system-arm cloud-image-utils
```

Comment:
question: other dependencies are installed from the `Makefile`, is it being done here as these aren't always needed?

Reply:
Is not needed in every case, but only if executing a build with qemu, so they are required to create a build environment (but not on travis for example).
`Makefile` install requirements inside the build environment (also on travis).

Reply:
That makes sense. I would suggest making this another Makefile target? I'm not a fan of apt-get running every time you use a script as it wastes time once the dependencies are installed.

Reply:
:+1: added `.installed-qemu` make target